graph/iterator/nodes.go

```diff
@@ -178,3 +178,100 @@ func (n *Nodes) Reset() {
 	n.pos = 0
 	n.iter = n.nodes.MapRange()
 }
+
+// NodesByEdge implements the graph.Nodes interfaces.
+// The iteration order of Nodes is randomized.
```

Comment:
Is it randomized deliberately, or as a side-effect?

Reply:
This is intentional thought we do provide iterators that are not randomised. I intend to put together a "Using Gonum Graphs" post for gonum.org that goes over some of this, but briefly, making graph iterators randomised reduces the possibility of accidentally relying on an iteration order in implementing a graph analysis. I've seen this in the wild, resulting in partial retraction of results.

Reply:
Neat.

Reply:
Do you have a test that they are indeed randomised?

Reply:
No, it's dependent on the runtime. I trust that.

Reply:
Also, I've had to write lot's of output conditioning code to make things pass because of the randomisation. So I am confident that it's there.